### PR TITLE
Allow UI validator to set constraints 

### LINKED ISSUE #2017: Validate after each field in a form
As a plugin author, 
Given that I am creating a form,
I would like there to be validation after each field is entered by the user
So that the user is aware of their mistakes before submitting the full form

Screen Shot 2021-02-24 at 1 11 49 PM

### PR DESCRIPTION
**Pull request by lenriquez**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**
- Fixes #2017

**Special notes for your reviewer**:
This PR allows to use all this validations but the last two
https://angular.io/api/forms/Validators

**Release note**:
```
release-note
```

### CODE REVIEW

**Comment by lenriquez** [comment]
@GuessWhoSamFoo Can you please help with this review?

**Comment by wwitzel3** [comment]
Does this actually change the existing behavior in anyway? This seems like it just changes the data format.

**Comment by GuessWhoSamFoo** [comment]
This allows validators like `min`, `max`, `pattern`. Typed strings can be used here to be more discoverable with what types of validation is available.

**Comment by wwitzel3** [comment]
Got it now, ok, yeah, I was just using the default validation for required and I was like, this seems the same, but now I get it. I think using Typed strings to expose the validation types is a great idea.

**Review by GuessWhoSamFoo — CHANGES_REQUESTED** [review]

File: web/src/app/modules/shared/components/presentation/form/form.component.ts [hunk 1]
```diff
@@ -68,15 +68,23 @@ export class FormComponent implements OnInit {
...
+    if (!validators) {
+      return [];
     }
-    return [];
+
+    const vFn: ValidatorFn[] = [];
+    for (let key in validators) {
+      const value = validators[key];
+
+      if (value === '') {
```
[the remark this answers is not in the document]
Comment (lenriquez) [reply web/src/app/modules/shared/components/presentation/form/form.component.ts:80]: I added checks for it, unfortunately every validation require at least one parameter, so I have to create a hash to check 
image

[the remark this answers is not in the document]
Comment (lenriquez) [reply web/src/app/modules/shared/components/presentation/form/form.component.ts:80]: image

**Review by wwitzel3 — APPROVED** [review]
LGTM, want to make sure @GuessWhoSamFoo is happy with it.

**Review by GuessWhoSamFoo — APPROVED** [review]
looks good - we need to remember this is a breaking API change and mark it as such in the changelog when releasing

**Pull request merged**